R/util.R

```diff
@@ -184,6 +184,78 @@ util.as_numeric_if_number <- function(x){
     return(x)
 }
 
+util.is_vector_of_logicals <- function(x){
+  # Are all elements of x logicals (permitting blanks, NOT permitting vectors entirely 
+  # composed of numeric booleans, i.e., 0/1)?
+  # (regardless of whether x is numeric)
+  # Character vectors sometimes need to be changed to logicals,
+  # e.g., when all columns set to character types by default.
+  
+  # if all the elements of x are numbers, return FALSE. This is to ensure correct handling of
+  # binary 0/1 variables as numbers, not logicals
+  if(util.is_vector_of_numbers(x)){
+    return(FALSE)
+  }
+  
+  # find logical values
+  x_as_logical <- suppressWarnings(as.logical(x))
+  
+  # anything that gets coerced to NA by as.numeric is not really a number.
```

Comment:
Nit: don't you mean "anything that gets coerced to NA by as.logical is not really a boolean"?